pkg/workloadendpointmanager/workloadendpointmanager_test.go

```diff
@@ -0,0 +1,337 @@
+)
+
+const namespace string = "group"
+const wename string = "workload"
+const containerID string = "dummy"
+
+var we *spiderpoolv1.WorkloadEndpoint
+var ipAllocation *spiderpoolv1.PodIPAllocation
+var HispodAllo []spiderpoolv1.PodIPAllocation
+var nodename string
+
+var _ = Describe("Workloadendpointmanager", Label("unittest", "WorkloadEndpoint"), func() {
+	BeforeEach(func() {
+		// Initialize parameter
+		nodename = "v1" + "node"
+
+		// Initialize PodIPAllocation
+		ipAllocation = &spiderpoolv1.PodIPAllocation{
+			ContainerID: containerID,
+			Node:        &nodename,
+			IPs: []spiderpoolv1.IPAllocationDetail{
+				{
+					NIC:         "eth0",
+					IPv4:        new(string),
+					IPv4Pool:    new(string),
+					IPv4Gateway: new(string),
+					IPv6:        new(string),
+					IPv6Pool:    new(string),
+					IPv6Gateway: new(string),
+				},
+			},
+		}
+		*ipAllocation.IPs[0].IPv4 = "127.1.0.5/24"
+		*ipAllocation.IPs[0].IPv4Pool = "pool1"
+		*ipAllocation.IPs[0].IPv4Gateway = "127.1.0.1"
+		*ipAllocation.IPs[0].IPv6 = "2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b/24"
+		*ipAllocation.IPs[0].IPv6Pool = "pool1"
+		*ipAllocation.IPs[0].IPv6Gateway = "127.2.3.1"
+
+		ipAllocationV2 := &spiderpoolv1.PodIPAllocation{
+			ContainerID: containerID,
+			Node:        &nodename,
+			IPs: []spiderpoolv1.IPAllocationDetail{
+				{
+					NIC:         "eth1",
+					IPv4:        new(string),
+					IPv4Pool:    new(string),
+					IPv4Gateway: new(string),
+					IPv6:        new(string),
+					IPv6Pool:    new(string),
+					IPv6Gateway: new(string),
+				},
+			},
+		}
+		*ipAllocationV2.IPs[0].IPv4 = "127.1.0.6/24"
+		*ipAllocationV2.IPs[0].IPv4Pool = "pool2"
+		*ipAllocationV2.IPs[0].IPv4Gateway = "127.1.0.2"
+		*ipAllocationV2.IPs[0].IPv6 = "3001:0db8:3c4d:0025:0000:0000:1a2f:1a2b/24"
+		*ipAllocationV2.IPs[0].IPv6Pool = "pool2"
+		*ipAllocationV2.IPs[0].IPv6Gateway = "127.2.3.2"
+
+		//HispodAllo := []spiderpoolv1.PodIPAllocation{podAllo, podAllo}
+		HispodAllo = []spiderpoolv1.PodIPAllocation{*ipAllocation, *ipAllocationV2}
+		//*podAllo.Node = nodename
+
+		// Initialize WorkloadEndpoint
+		we = &spiderpoolv1.WorkloadEndpoint{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      wename,
+				Namespace: namespace,
+			},
+			Status: spiderpoolv1.WorkloadEndpointStatus{
+				Current: ipAllocation,
+				History: HispodAllo,
+			},
+		}
+		// clean up the client
+		err = k8sClient.DeleteAllOf(ctx, we)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		err = k8sClient.Create(ctx, we)
+		Expect(err).ShouldNot(HaveOccurred())
+
+	})
+
+	Context("CURD operation in WorkloadEndpoint ", func() {
+		var newAllocation *spiderpoolv1.PodIPAllocation
+		BeforeEach(func() {
+			newAllocation = &spiderpoolv1.PodIPAllocation{
+				ContainerID: containerID,
+				Node:        &nodename,
+				IPs: []spiderpoolv1.IPAllocationDetail{
+					{
+						NIC: "eth0",
+					},
+					{
+						NIC: "eth1",
+					},
+				},
+			}
+
+			// update WorkloadEndpoint current
```

Comment:
```suggestion
			// update the current WorkloadEndpoint
```
Not sure, need your confirmation.